packages/grafana-data/src/dataframe/DataFrameJSON.ts

```diff
@@ -216,18 +231,34 @@ export function dataFrameToJSON(frame: DataFrame): DataFrameJSON {
   const data: DataFrameData = {
     values: [],
   };
+
+  const allNanos: Array<number[] | null> = [];
+  let hasNanos = false;
+
   const schema: DataFrameSchema = {
     refId: frame.refId,
     meta: frame.meta,
     name: frame.name,
     fields: frame.fields.map((f) => {
-      const { values, state, display, ...sfield } = f;
+      const { values, nanos, state, display, ...sfield } = f;
       delete (sfield as any).entities;
       data.values.push(values.toArray());
+
+      if (nanos != null) {
```

Comment:
the type of `nanos` is `number[] | undefined`, either the type or this condition is wrong

Reply:
`!= null` covers both `null` and `undefined` with fewer chars :)

Reply:
TIL _javascript magic_

```
> undefined != null
false
> undefined === null
false
> undefined == null
true
```

Anyway, it's just a `nit` but if `nanos` can be null I would define the type as `nanos?: number[] | null`

Reply:
sorry @andresmgot , it's an annoying trick (the `!= null`) and i in general really dislike such tricks, but this one is so good at basically `isNullish(value)` 😿